docs/src/content/concepts-certificates.md

```diff
@@ -198,16 +195,59 @@ openssl req -x509 -new -nodes -key ca.key -sha256 -out ca.crt -addext keyUsage=c
 cat ca.key ca.crt > mitmproxy-ca.pem
 ```
 
-## Using a client side certificate
+## Mutual TLS and client certificates
+
+Mutual TLS (mTLS) typically is used for the server to verify the client's
+identity, in addition to the more common validation of server identity (is the
+server the one they claim they are).
+
+If the server wants to verify the clients identity using mTLS, the server sends
+a `CertificateRequest` message to the client as part of the TLS handshake. The
+client then needs to provide its cert and proof ownership of the private key by
+generating a signature.
+
+Client certificates, just as the more commonly known server certificates, are
+full x509 certs with a public and private key, commonly stored as PEM
+certificate file.
+
+Some internet protocols use client and server certificates as a form of
+authentication and authorization, e.g., the MQTT protocol used in
+Internet-of-Things networks. To intercept such protocols using mitmproxy, both
+the client needs to asked to establish mTLS, and the upstream server needs to
+believe that mitmproxy is a mTLS-compliant device. Such protocols require both
+options outlined in the next sections for mitmproxy to work correctly.
+
+Example usage:
+
+```bash
+mitmproxy --set request_client_cert=True --set client_certs=my_certs_folder
+```
+
+with `my_certs_folder` containing an `mqtt.example.com.pem` x509 certificate
+(including private key). This will cause mitmproxy to use the provided cert only
+for requests to the upstream server at `mqtt.example.com`.
+
+### mTLS between client and mitmproxy
+
+To instruct mitmproxy to request a client certificate, you can pass the `--set
+request_client_cert=True` option to mitmproxy. This will generate the
+`CertificateRequest` TLS handshake message and (if successful) establish an mTLS
+connection.
+
+This option can be used together with the `client_certs` option to also
+establish an mTLS connection between mitmproxy and the upstream server. In this
```

Comment:
This is a bit misleading in the sense that we don't do any validation. So it's just TLS where we also ask the client nicely to maybe send a certificate. There is no authentication going on. 😛